process/rating/ratingsData.go

```diff
@@ -230,19 +250,16 @@ func (rd *RatingsData) EpochConfirmed(epoch uint32, _ uint64) {
 	)
 }
 
-func (rd *RatingsData) getMatchingVersion(currentEpoch uint32) ratingsStepsData {
-	// TODO: configuration slice can be sorted in a descending order and return when encountering an activation epoch older
-	// than the provided one
-	var ratingsStep ratingsStepsData
+func (rd *RatingsData) getMatchingVersion(epoch uint32) (ratingsStepsData, error) {
+	// the config values are sorted in descending order, so the matching version is the first one whose enable epoch is less or equal than the provided epoch
 	for _, ratingsStepConfig := range rd.ratingsStepsConfig {
-		if ratingsStepConfig.enableEpoch > currentEpoch {
-			continue
+		if ratingsStepConfig.enableEpoch <= epoch {
+			return ratingsStepConfig, nil
 		}
-
-		ratingsStep = ratingsStepConfig
 	}
 
-	return ratingsStep
+	// the code should never reach this point, since the config values are checked on the constructor
```

Comment:
indeed should never reach this point
it seems like a dead code but that's the way we need to make golang happy :)